scripts/solc-bin/bytecode_reports_for_modified_binaries.sh

```diff
@@ -0,0 +1,133 @@
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2020 solidity contributors.
+#------------------------------------------------------------------------------
+
+# FIXME: Can't use set -u because the old Bash on macOS treats empty arrays as unbound variables
+set -eo pipefail
+
+(( $# == 5 )) || { echo "ERROR: Not enough arguments."; exit 1; }
+
+platform="$1"
+base_ref="$2"
+top_ref="$3"
+solc_bin_dir="$4"
+solidity_dir="$5"
+
+report_dir="$PWD"
+tmp_dir=$(mktemp -d -t bytecode-reports-XXXXXX)
+solcjs_dir="$tmp_dir/solcjs"
+script_dir="$solidity_dir/scripts"
+
+cd "$tmp_dir"
+
+git clone https://github.com/ethereum/solc-js.git "$solcjs_dir"
+cd "$solcjs_dir"
+npm install
+rm soljson.js
+
+cd "${solc_bin_dir}/${platform}/"
+echo "Commit range: ${base_ref}..${top_ref}"
+
+modified_release_versions=$(
+    git diff --name-only "${base_ref}" |
+    sed -n -E 's/^[^\/]+\/(solc|soljson)-[0-9a-zA-Z-]+-v([0-9.]+)\+commit\.[0-9a-f]+(.[^.]+)?$/\2/p' |
+    sort -V |
+    uniq
+)
+echo "Release versions modified in the commit range:"
+echo "$modified_release_versions"
+
+# NOTE: We want perform the check when the soljson-* files in bin/ and wasm/ are modified too
+# because in that case the symlinks in emscripten-wasm32/ and emscripten-asmjs/ might remain
+# unchanged but were're assuming that these directories are never directly used as a platform name.
+[[ $platform != bin && $platform != wasm ]] || { echo "Invalid platform name."; exit 1; }
+
+platform_binaries="$(git ls-files "solc-${platform}-v*+commit.*" | sort -V)"
+
+for binary_name in $platform_binaries; do
+    solidity_version=$(echo "$binary_name" | sed -n -E 's/^solc-'"${platform}"'-v([0-9.]+)\+commit.+$/\1/p')
+
+    if echo "$modified_release_versions" | grep -x "$solidity_version"; then
+        echo "Binary ${binary_name} (version ${solidity_version}) matches one of the modified versions."
+
+        work_dir="${tmp_dir}/${binary_name}"
+        mkdir "$work_dir"
+        cd "$work_dir"
+
+        # While bytecode scripts come from the latest compiler, the test files should come from
+        # the Solidity version we're running them against to avoid errors due to breaking syntax changes.
+        git clone --branch "v${solidity_version}" "$solidity_dir" "${work_dir}/solidity/"
```

Comment:
Wouldn't shallow be enough here?

Reply:
`--depth` is not supported with local clones. I think git just hardlinks stuff inside `.git/` which is more efficient anyway.

Reply:
Ah, didn't know that (nor think about it) - but makes sense, yep!